src/back-end/lib/resources/affiliation.ts

```diff
@@ -173,7 +164,7 @@ const resource: Resource = {
 
   delete(connection) {
     return {
-      async validateRequestBody(request) {
+      async validateRequestBody(request): Promise<Validation<ValidatedDeleteRequestBody, DeleteValidationErrors>> {
```

Comment:
You shouldn't need to type the return value of this function, as it should be inferred by the Resource type above.

Reply:
Also, there is no permissions check for deleting?